R/particle_filter.R

```diff
@@ -394,6 +330,156 @@ particle_filter <- R6::R6Class(
   ))
 
 
+## This is the particle filter engine, which actually does the
+## running. It's not designed for end-user use, but will be used both
+## in running the particle filter normally and at a lower-level for
+## SMC2.
+particle_filter_run <- R6::R6Class(
+  "particle_filter_run",
+
+  private = list(
+    generator = NULL,
+    pars = NULL,
+    data = NULL,
+    steps = NULL,
+    n_particles = NULL,
+    n_threads = NULL,
+    initial = NULL,
+    index = NULL,
+    compare = NULL
+  ),
+
+  public = list(
+    model = NULL,
+    step_current = 0L,
```

Comment:
Should this be set to `steps[[1L]]`?

Reply:
Two issues here: first, that's not known at the point that this is set, as this is on the generator. Second, this is the index of the steps in `steps`! So the step might start on odin step 100, but that's the first step. So this is an index into `steps`. Ideas for a better name appreciated

Reply:
`current_step_index`?